#!/usr/bin/env python2

PACKAGE = "Vision"
# The following line should not be here for Groovy and above
import roslib;roslib.load_manifest(PACKAGE)

# For Groovy and above, the following line should be
# from dynamic_reconfigure.parameter_generator_catkin import *
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("minBuoyRadius",  int_t, 0, "Min buoy radius (pixels)", 6, 0, 200)
gen.add("contourMinArea", int_t, 0, "Min contour area (pixels)", 70000, 0, 100000)
gen.add("redHueLow",  int_t, 0, "Red Hue Low",    0, 0, 180)
gen.add("redHueHigh", int_t, 0, "Red Hue High",  20, 0, 180)
gen.add("redSatLow",  int_t, 0, "Red Sat Low",   64, 0, 255)
gen.add("redSatHigh", int_t, 0, "Red Sat High", 255, 0, 255)
gen.add("redValLow",  int_t, 0, "Red Val Low",   61, 0, 255)
gen.add("redValHigh", int_t, 0, "Red Val High", 255, 0, 255)

gen.add("redLedHueLow",  int_t, 0, "Red LED Hue Low",    0, 0, 180)
gen.add("redLedHueHigh", int_t, 0, "Red LED Hue High",  15, 0, 180)
gen.add("redLedSatLow",  int_t, 0, "Red LED Sat Low",    0, 0, 255)
gen.add("redLedSatHigh", int_t, 0, "Red LED Sat High", 135, 0, 255)
gen.add("redLedValLow",  int_t, 0, "Red LED Val Low",   70, 0, 255)
gen.add("redLedValHigh", int_t, 0, "Red LED Val High", 255, 0, 255)

gen.add("yellowHueLow",  int_t, 0, "Yellow Hue Low",    6, 0, 180)
gen.add("yellowHueHigh", int_t, 0, "Yellow Hue High",  40, 0, 180)
gen.add("yellowSatLow",  int_t, 0, "Yellow Sat Low",  128, 0, 255)
gen.add("yellowSatHigh", int_t, 0, "Yellow Sat High", 255, 0, 255)
gen.add("yellowValLow",  int_t, 0, "Yellow Val Low",   82, 0, 255)
gen.add("yellowValHigh", int_t, 0, "Yellow Val High", 255, 0, 255)

gen.add("greenHueLow",  int_t, 0, "Green Hue Low",   51, 0, 180)
gen.add("greenHueHigh", int_t, 0, "Green Hue High",  62, 0, 180)
gen.add("greenSatLow",  int_t, 0, "Green Sat Low",   44, 0, 255)
gen.add("greenSatHigh", int_t, 0, "Green Sat High",  72, 0, 255)
gen.add("greenValLow",  int_t, 0, "Green Val Low",   25, 0, 255)
gen.add("greenValHigh", int_t, 0, "Green Val High", 165, 0, 255)

gen.add("ledXMin", int_t, 0, "Min x for LED centroid", 270, 0, 640)
gen.add("ledXMax", int_t, 0, "Max x for LED centroid", 370, 0, 640)
gen.add("ledWMin", int_t, 0, "Min width for LED blob", 120, 0, 640)
gen.add("ledWMax", int_t, 0, "Max width for LED blob", 300, 0, 640)

gen.add("bumpTime", double_t, 0, "Duration in seconds for bump motion (single direction)", 3, 0, 10)
gen.add("bumpK", double_t, 0, "Forward setpoint for bump motion", 1, 0, 100)

gen.add("farForwardK", double_t, 0, "Forward K when far away", 24, 0, 100)
gen.add("farSideK", double_t, 0, "Sidemove K when far away", 40, 0, 100)

gen.add("nearForwardK", double_t, 0, "Forward K when near", 6, 0, 100)
gen.add("nearSideK", double_t, 0, "Sidemove K when near", 12, 0, 100)

exit(gen.generate(PACKAGE, "TrafficLight", "TrafficLight"))
